Feature/2 signup

log1995 opened the pull request.

log1995 committed:
회원가입 구현

log1995 committed:
회원가입 구현

log1995 committed:
회원가입 구현

log1995 committed:
회원가입 구현

f-lab-joseph commented:
각각의 커밋메세지를 의미있게 작성해주세요. git reset 에 대해서 공부해보시면 좋을 것 같은데 .. 
이건 나중에 보시고 다음 커밋 메세지부터 적용해주세요.

f-lab-joseph commented:
https://meetup.toast.com/posts/106

log1995 committed:
회원가입 구현
 - 아이디 중복 검사 구현
 - 회원가입 시, Auto Increment 값 자동으로 저장
 - 생성시간, 수정시간 저장

log1995 committed:
회원정보 받아오는 방식 수정
 - 기존 @Param 어노테이션에서 User 객체를 통해 받아오는 방식으로 수정

log1995 committed:
판매자 회원가입 구현

log1995 committed:
패스워드 정책 구현
 - 8자리 미만 입력 시 예외 발생

log1995 committed:
null 값에 대한 예외 발생 구현 및 기존 기능의 분리
 - 기존 : Service 클래스의 signUp 메서드를 통해 아이디 중복체크 및 비밀번호 자릿수 체크
 - 수정 : 아이디 중복체크 및 비밀 번호 자릿수 체크를 새로운 메서드로 분리
 - 메서드 이름과 구현기능이 일치해야한다고 생각하여 각각의 기능에 대해 메서드로 분리함

log1995 committed:
회원탈퇴 기능 구현

log1995 committed:
Merge remote-tracking branch 'MyRepository/feature/2-signup' into main

log1995 committed:
Merge remote-tracking branch 'origin/main'

# Conflicts:
#	build.gradle
#	src/main/resources/application.properties
